tests/e2e/steps_inactive_vals.go

```diff
@@ -0,0 +1,500 @@
+							ValidatorID("bob"):   200,
+							ValidatorID("carol"): 300,
+						},
+						StakedTokens: &map[ValidatorID]uint{
+							ValidatorID("alice"): 100000000, // but alice does not get jailed or slashed
+							ValidatorID("bob"):   200000000,
+							ValidatorID("carol"): 300000000,
+						},
+					},
+				},
+			},
+			// give carol more power so that she has enough power to validate if bob goes down
+			{
+				Action: DelegateTokensAction{
+					Chain:  ChainID("provi"),
+					From:   ValidatorID("carol"),
+					To:     ValidatorID("carol"),
+					Amount: 700000000, // carol needs to have more than 2/3rds of power(alice) + power(carol) + power(bob) to run both chains alone, so we stake some more to her
+				},
+				State: State{
+					ChainID("provi"): ChainState{
+						ValPowers: &map[ValidatorID]uint{
+							ValidatorID("alice"): 0,
+							ValidatorID("bob"):   200,
+							ValidatorID("carol"): 1000,
+						},
+						StakedTokens: &map[ValidatorID]uint{
+							ValidatorID("alice"): 100000000,
+							ValidatorID("bob"):   200000000,
+							ValidatorID("carol"): 1000000000,
+						},
+						// check that bob and carol get rewards, but alice does not
+						Rewards: &Rewards{
+							IsNativeDenom:       true, // check for rewards in the provider denom
+							IsIncrementalReward: true, // check rewards since block 1
+							IsRewarded: map[ValidatorID]bool{
+								ValidatorID("alice"): false,
+								ValidatorID("bob"):   true,
+								ValidatorID("carol"): true,
+							},
+						},
+					},
+				},
+			},
+			// bob goes offline
+			{
+				Action: DowntimeSlashAction{
+					Chain:     ChainID("provi"),
+					Validator: ValidatorID("bob"),
+				},
+				State: State{
+					ChainID("provi"): ChainState{
+						ValPowers: &map[ValidatorID]uint{
+							ValidatorID("alice"): 100, // alice gets into the active set
+							ValidatorID("bob"):   0,   // bob is jailed
+							ValidatorID("carol"): 1000,
+						},
+						StakedTokens: &map[ValidatorID]uint{
+							ValidatorID("alice"): 100000000,
+							ValidatorID("bob"):   198000000, // 1% slash
+							ValidatorID("carol"): 1000000000,
+						},
+					},
+				},
+			},
+			{
+				// relay packets so that the consumer gets up to date with the provider
+				Action: RelayPacketsAction{
+					ChainA:  ChainID("provi"),
+					ChainB:  ChainID("consu"),
+					Port:    "provider",
+					Channel: 0,
+				},
+				State: State{
+					ChainID("provi"): ChainState{
+						// check that now every validator got rewarded since the first block
+						Rewards: &Rewards{
+							IsNativeDenom:       true, // check for rewards in the provider denom
+							IsIncrementalReward: true, // check rewards for currently produced blocks only
+							IsRewarded: map[ValidatorID]bool{
+								ValidatorID("alice"): true,  // alice is participating right now, so gets rewards
+								ValidatorID("bob"):   false, // bob does not get rewards since he is not participating in consensus
```

Comment:
Similar to my previous comment, here bob is not `IsRewarded` but above it is written that "every validator got rewarded".